File: app/components/folio/console/tiptap/simple_form_wrap/unsaved_changes_component.rb
# frozen_string_literal: true

class Folio::Console::Tiptap::SimpleFormWrap::UnsavedChangesComponent < Folio::Console::ApplicationComponent
  attr_reader :object

  def initialize(object:)
    @object = object
  end

  def data
    stimulus_controller("f-c-tiptap-simple-form-wrap-unsaved-changes")
  end

  private
    def latest_revision_info
      latest_revision = object.latest_tiptap_revision(user: Folio::Current.user)
      "#{l(latest_revision.created_at, format: :short)} – #{latest_revision.user.full_name}"
    end
end

File: app/components/folio/console/tiptap/simple_form_wrap/unsaved_changes_component.js
window.Folio.Stimulus.register('f-c-tiptap-simple-form-wrap-unsaved-changes', class extends window.Stimulus.Controller {
  static values = {
    placementType: String,
    placementId: Number,
    deleteUrl: String,
  }

  onContinueButtonClick () {
    this.element.style.display = 'none'
    this.dispatch('continueUnsavedChanges', { bubbles: true })
  }

  onDiscardButtonClick () {
    const data = {
      placement: {
        type: this.placementTypeValue,
        id: this.placementIdValue
      }
    }

    window.Folio.Api.apiDelete(this.deleteUrlValue, data)
      .then((response) => {
        if (response && response.success) {
          window.location.reload()
        }
      })
      .catch((error) => {
        console.warn('[Folio] [Tiptap] Discard failed:', error)
      })
  }
})
